src/attributes/codegen.md

```diff
@@ -140,6 +140,28 @@ feature detection on the x86 platforms.
 > may be enabled or disabled for an entire crate with the
 > [`-C target-feature`] flag.
 
+## The `track_caller` attribute
+
+The `track_caller` attribute applies to function declarations, allowing code within the function to
+observe the callsite of the topmost `track_caller`-attributed function in the attributed function's
+callers. If the attributed function itself is called by an unattributed function, then the
+attributed function observes its own callsite. If the attributed function is called by another
+attributed function, then code within the callee observes the callsite of the caller, and so on.
```

Comment:
General note re. this paragraph: There's a lot of "attributed function" which makes things feel somewhat dense to follow. I think we could attach some symbols to the cases to make the references in the text clearer (requiring less mental state). For example:
> If the attributed function `f` is called by another attributed function `g`, then code within the `f` observes the callsite of `g`, and so on.

Reply:
Question: does the reference have a notion of the call stack?

Reply:
I ask because I think miri's implementation of walking up the call stack is actually the most intuitive way to understand the semantics of the attribute. Describing it is pretty natural in terms of the observation point "looking up" the stack until it finds an unattributed function's frame.

Reply:
I "grepped" for "stack" and "call" and found the following:
- https://github.com/rust-lang/reference/blob/8e7d614303b0dec7492e048e63855fcd3b944ec8/src/expressions/return-expr.md#L9

- https://github.com/rust-lang/reference/blob/33dfdbda87931a8e0868a6c5066cc272463c8b7e/src/memory-ownership.md#L3

which seems like a "yes"?

I agree that what you suggest would be intuitive.